### ISSUE TITLE
migrations occasionally shit the bed

### ISSUE THREAD

**Issue by cultofmetatron** [knex/knex#312]
We just switched to .6 and the migrations are acting "janky"

TLDR: I run the test, it works, I run it 2 or 3 times and suddenly it throws errors all over the place and the migrations table is sudden;y empty as if the records are deleted. on average, there's a 50% chance the migration will fail necessitating the workaround described below.

we have some tests that do a full rollback and migrate before running

``` javascript
db.knex.migrate.rollback(knexFile)
   .then(function() {
      return db.knex.migrate.latest(knexFile);
    })
```

I cleared my database and ran the test and it seemed to work fine, however after doing it 3-4 times, the system craps out royally.

the error looks like 

```
Possibly unhandled error: relation "roles" does not exist
    at Connection.parseE (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:526:11)
    at Connection.parseMessage (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:356:17)
    at Socket.<anonymous> (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:105:22)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:746:14)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:408:10)
    at emitReadable (_stream_readable.js:404:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
    at TCP.onread (net.js:528:21)
```

I run it again, and it says 

```
Possibly unhandled error: relation "user_company" already exists
    at Connection.parseE (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:526:11)
    at Connection.parseMessage (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:356:17)
    at Socket.<anonymous> (/Users/cultofmetatron/keychain/quintesson/node_modules/pg/lib/connection.js:105:22)
    at Socket.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:746:14)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:408:10)
    at emitReadable (_stream_readable.js:404:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
    at TCP.onread (net.js:528:21)
```

The strangest part is that when I look at the migrations table, its suddenly empty.  my current workaround is to drop all tables and rerun it after a quick prayer to chutulu, satan and any other deity that'll listen.
